filewatcher/filewatcher.go

```diff
@@ -81,22 +94,16 @@ func (r *Result) Stop() {
 	r.cancel()
 }
 
-func (r *Result) watcherLoop(
-	watcher *fsnotify.Watcher,
-	path string,
-	parser Parser,
-	softLimit, hardLimit int64,
-	logger log.Wrapper,
-) {
-	file := filepath.Base(path)
+func (r *Result) watcherLoop(watcher *fsnotify.Watcher) {
+	file := filepath.Base(r.path)
 	for {
 		select {
 		case <-r.ctx.Done():
 			watcher.Close()
 			return
 
 		case err := <-watcher.Errors:
-			logger.Log(context.Background(), "filewatcher: watcher error: "+err.Error())
+			r.logger.Log(context.Background(), "filewatcher: watcher error: "+err.Error())
 
 		case ev := <-watcher.Events:
```

Comment:
When I was suggesting polling, I was imagining a new constructor that specified an additional fixed polling period, plus some sensible default (e.g. 30 seconds) for the existing ones, not checking on every Get call.

WDYT?

Reply:
I certainly like that better as it makes things much simpler, let me ask internally on what's an appropriate default.